src/netaddress.cpp

```diff
@@ -341,9 +329,9 @@ enum Network CNetAddr::GetNetwork() const
 std::string CNetAddr::ToStringIP() const
 {
     if (IsTor())
-        return EncodeBase32(&ip[6], 10) + ".onion";
+        return EncodeBase32(m_addr.data(), m_addr.size()) + ".onion";
     if (IsInternal())
-        return EncodeBase32(ip + sizeof(g_internal_prefix), sizeof(ip) - sizeof(g_internal_prefix)) + ".internal";
+        return EncodeBase32(m_addr.data(), m_addr.size()) + ".internal";
```

Comment:
Heh we need to make this one take a span too (not in this PR ofcourse).

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/19687